src/scala/com/github/johnynek/bazel_deps/S3URLHandler.scala

```diff
@@ -0,0 +1,422 @@
+package com.github.johnynek.bazel_deps
+
+import java.io.{File, FileInputStream, InputStream}
+import java.net.{InetAddress, URI, URL}
+import java.util.Properties
+import java.util.concurrent.ConcurrentHashMap
+import com.amazonaws.ClientConfiguration
+import com.amazonaws.SDKGlobalConfiguration.{ACCESS_KEY_ENV_VAR, ACCESS_KEY_SYSTEM_PROPERTY, SECRET_KEY_ENV_VAR, SECRET_KEY_SYSTEM_PROPERTY}
+import com.amazonaws.auth._
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.s3.model._
+import com.amazonaws.services.s3.{AmazonS3, AmazonS3Client, AmazonS3URI}
+import com.amazonaws.services.securitytoken.{AWSSecurityTokenService, AWSSecurityTokenServiceClient}
+import com.amazonaws.services.securitytoken.model.{AssumeRoleRequest, AssumeRoleResult}
+import org.apache.ivy.util.url.URLHandler
+import org.apache.ivy.util.{CopyProgressEvent, CopyProgressListener, Message}
+import scala.collection.JavaConverters._
+import scala.util.Try
+import scala.util.matching.Regex
+
+object S3URLHandler {
+  private val DOT_SBT_DIR: File = new File(System.getProperty("user.home"), ".sbt")
```

Comment:
I'm concerned this `.sbt` reference means this is still rather coupled with sbt.